src/index.js

```diff
@@ -2,42 +2,141 @@ import '@fortawesome/fontawesome-free/css/all.css';
         <span class="dots">
+        <span class="close-task" id="${list.index}">
+          <i class="fa-solid fa-rectangle-xmark"></i>
+        </span>
+        <span class="edit-task" id="${list.index}">
         <i class="fa-solid fa-ellipsis-vertical"></i>
+        </span>
       </span></li>`;
-  document.getElementById('list-block').innerHTML += card;
+    document.getElementById('list-block').innerHTML += card;
+  });
+
+  // click event variables
+
+  const removeTasks = document.querySelectorAll('.close-task');
+  const editTasks = document.querySelectorAll('.edit-task');
+  const checkBoxes = document.querySelectorAll('.check');
+
+  // checkbox functionality when box is click check appears
+
+  checkBoxes.forEach((checkBox) => {
+    checkBox.addEventListener('change', (e) => {
+      e.preventDefault();
+      const check = checkBox.getAttribute('id') * 1;
+      const localStorage = new Storage();
+      const lists = localStorage.getStorage();
+      if (checkBox.checked) {
+        lists[check - 1].completed = true;
+      } else {
+        lists[check - 1].completed = false;
+      }
+      localStorage.localStorage(lists);
+      updateList();
+    });
+  });
+
+  // edit task inside the list by using prompt when the 3 dots are clicked
+
+  editTasks.forEach((editTask) => {
+    editTask.addEventListener('click', (e) => {
+      e.preventDefault();
+      const localStorage = new Storage();
+      const lists = localStorage.getStorage();
+      const task = editTask.getAttribute('id') * 1 - 1;
+      const newTask = prompt('Change task');
+      lists[task].description = newTask;
+      localStorage.localStorage(lists);
+      updateList();
+    });
+  });
+
+  // removes task from the list
+
+  removeTasks.forEach((removeTask) => {
+    removeTask.addEventListener('click', (e) => {
+      e.preventDefault();
+      const localStorage = new Storage();
+      const index = removeTask.getAttribute('id') * 1 - 1;
+      const lists = localStorage.getStorage();
+      const taskDelete = lists.filter((item, key) => {
+        if (key !== index) {
+          return true;
+        }
+        return null;
+      });
+      for (let i = 0; i < taskDelete.length; i += 1) {
+        taskDelete[i].index = i + 1;
+      }
+      localStorage.localStorage(taskDelete);
+      updateList();
+    });
+  });
+};
+updateList();
+
+const lists = [];
+lists.sort((a, b) => a.index - b.index);
+
+// Add task click event
+
+const AddtoList = document.querySelector('.add-button');
+
+AddtoList.addEventListener('click', (e) => {
+  e.preventDefault();
+  const localStorage = new Storage();
+  const lists = localStorage.getStorage();
+  const lastIndex = lists.length + 1;
+  lists.push({
+    description: document.getElementById('add').value,
+    completed: false,
+    index: lastIndex,
+  });
+  localStorage.localStorage(lists);
+  updateList();
+});
```

Comment:
- [ **OPTIONAL** ] I think it would be better if you also add a feature to clean your input text after adding a new to-do task to make it clean. I know you can do it :muscle: :muscle: